### PR TITLE
[Docs] Docs for multiple resources interface

### PR DESCRIPTION
**Pull request by Michaelvll**
This is to add a more detailed doc for multiple resources interface added by #2498

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/auto-failover.rst [hunk 1]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
+    accelerators: [A10:1, A10g:1, L4:1]
+
+In the above example, SkyPilot will first try to provision an A10 GPU, then an A10g GPU, and finally an L4 GPU:
+:code:`sky launch task.yaml`
+
+
+(**Advanced**) Multiple Candidate Resources
+--------------------------------------------
+
+If a task would like to specify multiple candidate resources (not only GPUs), the user can specify a list of candidate resources with a preference annotation:
```
Comment (MaoZiming) [inline docs/source/examples/auto-failover.rst:127]: Might be good to explain the syntax a bit more. Something like:
`-` marks one candidate resource
Use indent to include properties of that candidate resource (e.g. `accelerators`, `instance_type`, etc.)

**Review by MaoZiming — APPROVED** [review]
Looks good to me! Thanks for writing up the doc

File: docs/source/examples/auto-failover.rst [hunk 2]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
+          region: eastus
+          accelerator: A100
+          
+This will be useful when only a set of regions/clouds are desired for launching the resources:
+
+.. code-block:: yaml
+
+  resources:
+    any_of:
+      - region: us-east-1
```
Comment (MaoZiming) [inline docs/source/examples/auto-failover.rst:159]: This example is a bit complicated. Some regions are in AWS and some in GCP. Maybe simplify the example a bit

File: docs/source/examples/auto-failover.rst [hunk 3]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
+    accelerators: {A10:1, L4:1, A10g:1}
+
+In the above example, SkyPilot will try to provision the any cheapest available GPU within the set of
+A10, L4, and A10g GPUs, using :code:`sky launch task.yaml`.
+
+.. code-block:: console
+
+  $ sky launch task.yaml
+  ...
+  -----------------------------------------------------------------------------------------------------
```
Comment (romilbhardwaj) [inline docs/source/examples/auto-failover.rst:103]: Should we have the same indent for this table row separator too? Also, might be fine to remove the logging line (i.e., `SKYPILOT_MINIMIZE_LOGGING=1`)
```suggestion
I 11-19 08:00:03 optimizer.py:910]  -----------------------------------------------------------------------------------------------------
```

Same below

File: docs/source/examples/auto-failover.rst [hunk 4]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
   I 02-23 16:42:26 cloud_vm_ray_backend.py:668]
   I 02-23 16:42:26 cloud_vm_ray_backend.py:668] Launching on AWS us-west-2 (us-west-2a,us-west-2b,us-west-2c,us-west-2d)
   I 02-23 16:47:04 cloud_vm_ray_backend.py:740] Successfully provisioned or found existing VM. Setup completed.
+
+
+Multiple Candidate GPUs
+-------------------------
+
+If a task can be run on different GPUs, the user can specific multiple candidate GPUs,
+and SkyPilot will automatically find the best available GPU.
```
Comment (romilbhardwaj) [inline docs/source/examples/auto-failover.rst:87]: Should we clarify the optimization objective? Users may confuse best for some kind of automatic GPU selection which we don't do

```suggestion
and SkyPilot will automatically find the cheapest available GPU.
```

File: docs/source/examples/auto-failover.rst [hunk 5]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
+  ...
   I 02-23 16:42:26 cloud_vm_ray_backend.py:668]
   I 02-23 16:42:26 cloud_vm_ray_backend.py:668] Launching on AWS us-west-2 (us-west-2a,us-west-2b,us-west-2c,us-west-2d)
   I 02-23 16:47:04 cloud_vm_ray_backend.py:740] Successfully provisioned or found existing VM. Setup completed.
+
+
+Multiple Candidate GPUs
+-------------------------
+
+If a task can be run on different GPUs, the user can specific multiple candidate GPUs,
```
Comment (romilbhardwaj) [inline docs/source/examples/auto-failover.rst:86]:
```suggestion
If a task can be run on different GPUs, the user can specify multiple candidate GPUs,
```

File: docs/source/examples/auto-failover.rst [hunk 6]
```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
...
+
+To specify a preference order, use a list of candidate GPUs in the task yaml:
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: [A10:1, A10g:1, L4:1]
+
+In the above example, SkyPilot will first try to provision an A10 GPU, then an A10g GPU, and finally an L4 GPU:
+:code:`sky launch task.yaml`
```
Comment (romilbhardwaj) [inline docs/source/examples/auto-failover.rst:121]: Perhaps mentioning `sky launch` is not required here

```suggestion
In the above example, SkyPilot will first try to provision an A10 GPU, then an A10g GPU, and finally an L4 GPU.
```

File: docs/source/examples/auto-failover.rst (hunk 1 above)
Comment (romilbhardwaj) [inline docs/source/examples/auto-failover.rst:127]: Nit - wondering if we can include a short snippet for the python API as well :)

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll!

File: docs/source/examples/auto-failover.rst (hunk 3 above)
Comment (Michaelvll) [reply docs/source/examples/auto-failover.rst]: Thanks for catching this! I tried `SKYPILOT_MINIMIZE_LOGGING` here, but the lines are a bit hard to distinguish from each other, as the color cannot be shown in the block. I make the output align with the prefix instead. : )

File: docs/source/examples/auto-failover.rst (hunk 2 above)
Comment (Michaelvll) [reply docs/source/examples/auto-failover.rst]: Just reduced the candidate to only two. PTAL : )

File: docs/source/examples/auto-failover.rst (hunk 1 above)
Comment (Michaelvll) [reply docs/source/examples/auto-failover.rst:126]: Good point! I added a tip block. How does that look?

Comment (Michaelvll) [reply docs/source/examples/auto-failover.rst:126]: I feel like we may need a more thorough modification of the doc to add python API, i.e., adding it in many other pages. Maybe we can add it in a future PR?

**Review by romilbhardwaj — APPROVED** [review]
Looks great - thanks @Michaelvll!

**Pull request merged**